site/docs/plugins/keyboard.md

```diff
@@ -273,6 +273,17 @@ await ctx.reply(text, {
 });
 ```
 
+### Responding to Clicks
+
+As mentioned earlier, all that keyboards do is sending regular text messages.
```

Comment:
Maybe it should be clarified that this is about the custom keyboards and not the inline keyboards?

Reply:
We already have a section with the exact same title under the inline keyboard section.

Reply:
"all that keyboards" still can sound misleading.

Reply:
I see that point. I'm not sure how to do about this, I don't want to say “keyboards (not inline keyboards)” everywhere

Reply:
What do you think about b7a28ef7c4932faf74abb84d8c363180ccf434ee?

Reply:
I remember them being referred to as "custom keyboards" in the documentation which seems more declarative than just "keyboard". Looking into the documentation its not always the case but its definitely done in many cases.

Reply:
The Bot API types use `Keyboard`. They may also use `ReplyKeyboard`. Be it custom keyboard or reply keyboard, I would not want to introduce a _third_ term. If we rename the `Keyboard` class, I would be open to using a different term consistently everywhere.

Reply:
The type was Keyboard from the beginning without the knowledge of another appearing later. I dont think the types itself should be used as a single reference of what its called. As it is referred to as custom keyboard I would do that too as it clarifies a lot what is meant by "keyboard". This doesnt need to change the typings as they are mirroring the Telegram types but can change how they are referred in the docs. The goal of the grammY docs is to be easy to understand and a "keyboard" can differ with its meaning.

But just because I would approach it that way doesnt mean its the only/best way to go.

Reply:
I don't like about the term custom keyboards that it makes it sound like inline keyboard could not be custom-built. I think that can be confusing. Do you agree?

I don't like about the term reply keyboard that we already overload the word reply thrice (`reply_to_message_id`, `ctx.reply`, `reply_markup`) and I am not comfortable adding a fourth meaning. What do you think?

Overall, I am more comfortable with calling it custom keyboards and then casually describing inline keyboards as custom-built somewhere, rather than calling keyboards from now on reply keyboards.

Reply:
I don't think "Reply Keyboard" would be a good term for it. `sendMessage()` can also send that keyboard without being a reply. Which is also true for `reply_markup`. (Thinking about it… Telegraf named the class Markup and not ReplyMarkup which seems to make more sense. It might also because that name is shorter but still.)

About the naming and how I interpreted it so far:
You have your normal phone keyboard. It can be replaced with a custom keyboard typing predefined, custom texts instead of the letters the normal phone keyboard provides.
Also there is another keyboard attached to messages, with is in between the messages/lines → inline.

> it makes it sound like inline keyboard could not be custom-built

Hm… I guess I understand the thought behind it but doesnt seem very confusing to me. (Which doesnt mean it is a less valid argument.) Personally I think the inline_query and inline_keyboard stuff is more confusing. While the inline_keyboard is the the only thing possible for inline query results the inline keyboard also works independent of the inline query / inline mode.
I dont think we will get rid of every confusing aspect of the API but as long as we can describe everything by their distinct names we can explain things.

I think calling something A and something else BA while aspects of A dont work for BA and the other way around is more confusing than calling something with distinct CA and BA names. (This is the same problem with inheritance of classes which easily become inconsistent)
In the case of "custom keyboard" we wouldnt invent a new term as the official Bot API docs already call it that way.

Reply:
Alright. Then let's switch to calling them custom keyboards.

This requires the entire documentation to be updated, which includes
- this article
- all references to this article
- all references to header links (the titles will change)
- all TSDoc in the grammY core package


I can do this, but it may take some time.

Reply:
Not something that has to be updated all at the same time. Also something that can be done by someone willing to improve docs. Maybe create a separate Issue for that so others can see it being open and provide PRs more easily?

Reply:
I prefer very much to do all changes immediately and consistently. Nothing is worse than outdated docs. Please see https://github.com/grammyjs/grammY/pull/135